### ISSUE TITLE
Error: OTP hash or nand key are invalid

### ISSUE THREAD

**Issue by Dimensional** [AuroraWright/SafeA9LHInstaller#26]
I was attempting to update with the version 2.6.1 installer, but it is giving that error. I'm attempting to upgrade on my n3DS. I have the OTP file on the SD card, having never removed it when I did the 2.1 downgrade guide, and I compared it to backups I made, with it matching all 3 checksums I used; MD5, SHA-256 and SHA-512. Could the installer have preference to OTP-less installation even with the presence of the OTP file?

Addendum: Attempted to do the same update on my o3DS, and same error. Again, tested checksums and they all matched.

**Comment by Dimensional** [comment]
This is still happening on 2.6.2. I don't understand why it's happening like that, and how to fix it. Any suggestions?

**Comment by AuroraWright** [comment]
Your stage2 doesn't preserve the contents of the SHA register. Which do you have installed?

**Comment by Dimensional** [comment]
I had updated to A9LHv2 back in June from your repos.

**Comment by Dimensional** [comment]
I guess I didn't understand the question?

**Comment by Dimensional** [comment]
I decided to bypass luma and load the updater directly, and the installation worked. I typically update Luma to the nightly, but I attempted to install with the stable version too, and it was when I was having trouble installing things through FreeShop I learned that the nightly was somehow causing issues with the keys, giving me the idea that Luma could be causing problems with other stuff. And testing on both my n3DS and o3DS, I'm able to confirm the problem is with loading through Luma, not the updater directly.
